spring-cloud-stream/src/main/java/org/springframework/cloud/stream/config/ChannelBindingAutoConfiguration.java

```diff
@@ -40,6 +45,7 @@
 @Configuration
 @ConditionalOnBean(ChannelBindingService.class)
 @EnableConfigurationProperties(DefaultPollerProperties.class)
+@AutoConfigureBefore(EndpointAutoConfiguration.class)
```

Comment:
In the current state of the PR, this is not necessary, since the health infrastructure is created in a separate configuration - this is a leftover from an earlier approach. However, now that the question has been asked, I'd be much more keen on moving the health endpoint bean here. The rationale of having it close to the other Binder infrastructure beans was to have all the binder-related functionality in one place. However, I'd rather think that the autoconfig should have everything that relates to Boot autoconfig (including health endpoints).